### PR TITLE
Add keybinding for toggling outlier detection

### LINKED ISSUE #46: Move curve-fitting.model/get-coeffs-from-trace into curve-fitting.model.trace
It's not a part of the model, after all. Will require rewriting it in Clojure.

### LINKED ISSUE #57: Demo crashes when outliers are disabled
Looks like I merged #50 prematurely. 

## Reproduction steps
Change
```clojure
(def outliers-enabled false)
```
to
```clojure
(def outliers-enabled false)
```
in `model.clj`, run `(go)` in the REPL and click anywhere in the display window.

## Expected results
The point appears in the display window.

## Actual results
```
java.lang.AssertionError: Assert failed: ["no such subtrace" 2 {"gaussian" {:value -152/25}}]
(not (= val nil))
	at metaprob.state$subtrace.invokeStatic(state.cljc:65)
	at metaprob.state$subtrace.invoke(state.cljc:58)
	at metaprob.trace$trace_direct_subtrace.invokeStatic(trace.cljc:183)
	at metaprob.trace$trace_direct_subtrace.invoke(trace.cljc:182)
	at metaprob.trace$trace_subtrace.invokeStatic(trace.cljc:201)
	at metaprob.trace$trace_subtrace.invoke(trace.cljc:196)
	at curve_fitting.model.trace$outliers$fn__11585.invoke(trace.clj:41)
	at clojure.core$map$fn__5587.invoke(core.clj:2745)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:528)
	at clojure.core$seq__5124.invokeStatic(core.clj:137)
	at clojure.core$seq__5124.invoke(core.clj:137)
	at clojure.core$map$fn__5587.invoke(core.clj:2747)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:528)
	at clojure.core$seq__5124.invokeStatic(core.clj:137)
	at clojure.core$every_QMARK_.invokeStatic(core.clj:2671)
	at clojure.core$map$step__5598$fn__5599.invoke(core.clj:2764)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:528)
	at clojure.core$seq__5124.invokeStatic(core.clj:137)
	at clojure.core$map$fn__5587.invoke(core.clj:2738)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:528)
	at clojure.core$seq__5124.invokeStatic(core.clj:137)
	at clojure.core$map$fn__5587.invoke(core.clj:2738)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:528)
	at clojure.core$seq__5124.invokeStatic(core.clj:137)
	at clojure.core$map$fn__5591.invoke(core.clj:2749)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:528)
	at clojure.core$seq__5124.invokeStatic(core.clj:137)
	at clojure.core$seq__5124.invoke(core.clj:137)
	at curve_fitting.core$draw_clicked_points_BANG_.invokeStatic(core.clj:71)
	at curve_fitting.core$draw_clicked_points_BANG_.invoke(core.clj:60)
	at curve_fitting.core$draw_BANG_.invokeStatic(core.clj:148)
	at curve_fitting.core$draw_BANG_.invoke(core.clj:127)
	at curve_fitting.sketches$applet$fn__13365.invoke(sketches.clj:66)
	at quil.middlewares.fun_mode$wrap_draw_update$quil_draw__1721.invoke(fun_mode.cljc:16)
	at quil.middlewares.safe_fns$wrap_fn$fn__114.invoke(safe_fns.clj:8)
	at quil.applet$_draw.invokeStatic(applet.clj:212)
	at quil.applet$_draw.invoke(applet.clj:210)
	at quil.Applet.draw(Unknown Source)
	at processing.core.PApplet.handleDraw(PApplet.java:2426)
	at quil.Applet.handleDraw(Unknown Source)
	at processing.awt.PSurfaceAWT$12.callDraw(PSurfaceAWT.java:1557)
	at processing.core.PSurfaceNone$AnimationThread.run(PSurfaceNone.java:316)
```

### PR DESCRIPTION
**Pull request by zane**
Fixes #46.
Fixes #57.

### CODE REVIEW

File: src/curve_fitting/model.clj [hunk 1]
```diff
@@ -26,37 +24,18 @@
...
 
-(define outlier?
-  (gen []
-    (and outliers-enabled
-         (flip 0.1))))
-
 (define add-noise-to-curve
   (gen [curve]
+    ;; Outliers are disabled by default. Enable with an intervention trace.
+    (define outliers-enabled? (flip 1)) ; Are outliers enabled?
```
Comment (joshuathayer) [inline src/curve_fitting/model.clj:30]: Without an intervention one way or the other, won't this enable outliers half the time?

Comment (zane) [reply src/curve_fitting/model.clj:30]: The argument to `flip` is the probability that it returns `true`, so outliers should be enabled every time if no intervention trace is provided.

Comment (zane) [reply src/curve_fitting/model.clj:30]: I should update that comment.

File: src/curve_fitting/model.clj [hunk 2]
```diff
@@ -26,37 +24,18 @@
...
                      (range degree))))))
 
-(define outlier?
-  (gen []
-    (and outliers-enabled
-         (flip 0.1))))
-
 (define add-noise-to-curve
   (gen [curve]
+    ;; Outliers are disabled by default. Enable with an intervention trace.
```
Comment (joshuathayer) [inline src/curve_fitting/model.clj]:
```suggestion
    ;; Outliers are enabled by default. Enable with an intervention trace.
```

**Pull request merged**